plugins/inputs/ctrlx_datalayer_sse/ctrlx_datalayer_sse_test.go

```diff
@@ -0,0 +1,296 @@
+package ctrlx_datalayer_sse
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCtrlXGetAuthTokenBasic(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		_, err := w.Write([]byte("{\"access_token\": \"eyJhbGciOiJIU.xxx.xxx\", \"token_type\":\"Bearer\"}"))
+		require.NoError(t, err)
+	}))
+	defer server.Close()
+
+	s := &ctrlXDataLayerSSE{
+		connection:   &http.Client{},
+		Hostname:     server.URL,
+		Username:     "user",
+		Password:     "password",
+		Subscription: nil,
+		Log:          testutil.Logger{},
+	}
+
+	s.setAuthToken()
+
+	if s.token.AccessToken != "eyJhbGciOiJIU.xxx.xxx" {
+		t.Errorf("expect access_toke 'eyJhbGciOiJIU.xxx.xxx', but get %v\n", s.token.AccessToken)
+	}
+	if s.token.TokenType != "Bearer" {
+		t.Errorf("expect token_type 'Bearer', but get %v\n", s.token.TokenType)
+	}
+}
+
+func TestCtrlXGetAuthTokenMatrixDriven(t *testing.T) {
+	var tests = []struct {
+		res   string
+		token token
+	}{
+		{res: "{\"status\":400}", token: token{AccessToken: "", TokenType: ""}},
+		{res: "{\"status\":500}", token: token{AccessToken: "", TokenType: ""}},
+	}
+
+	for _, test := range tests {
+		t.Run(test.res, func(t *testing.T) {
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				_, err := w.Write([]byte(test.res))
+				require.NoError(t, err)
+			}))
+			defer server.Close()
+			s := &ctrlXDataLayerSSE{
+				connection:   &http.Client{},
+				Hostname:     server.URL,
+				Username:     "user",
+				Password:     "password",
+				Subscription: nil,
+				Log:          testutil.Logger{},
+			}
+			s.setAuthToken()
+			if s.token.AccessToken != test.token.AccessToken {
+				t.Errorf("test %v: expect access_toke %v, but get %v\n", test.res, test.token.AccessToken, s.token.AccessToken)
+			}
+			if s.token.TokenType != test.token.TokenType {
+				t.Errorf("test %v: expect token_type %v, but get %v\n", test.res, test.token.TokenType, s.token.TokenType)
+			}
+		})
+	}
+}
+
+var server *httptest.Server
```

Comment:
Please don't make this a global variable, but keep it locally in the tests. This avoids potential side-effects between the tests.

Reply:
Done